### PR TITLE
fix(skore/CrossValidationReport): Raise `NotImplementedError` upon `data_source="both"`

### LINKED ISSUE #2546: For CV reports, `report.metrics.summarize(data_source="both")` returns misleading results.
### Describe the bug

From the docstring, I read that `data_source="both"` should not be accepted for `CrossValidationReport.metrics.summarize`, but it actually is and returns wrong results (unweighted average of train & test, which doesn't make sense and is not something you want to look at).

I think it would make sense to accept it and to return something similar what's returned for `EstimatorReport` (concatenation of metrics of both sources):

| Metric           |   RandomForestRegressor (train) |   RandomForestRegressor (test) |
|:-----------------|--------------------------------:|-------------------------------:|
| R²               |                       0.803981  |                      0.807788  |
| RMSE             |                       0.219989  |                      0.22171   |
| Fit time (s)     |                       0.23386   |                      0.23386   |
| Predict time (s) |                       0.0230818 |                      0.0231872 |

### Steps/Code to Reproduce

```Python
import numpy as np

from sklearn.ensemble import RandomForestRegressor
from skore import CrossValidationReport, EstimatorReport

n = 10_000
X = np.random.rand(n, 5)
y = X[:, :3].sum(axis=1)

reg = RandomForestRegressor(random_state=0, min_impurity_decrease=0.01, n_estimators=50, n_jobs=2)
report = CrossValidationReport(reg, X, y)
report.metrics.summarize(data_source="both").frame()
```

### Expected Behavior

Concat the data of both sources

### Actual Behavior

Averages the data of both sources

### Environment

```shell
System:
    python: 3.12.11 (main, Aug 18 2025, 19:19:11) [Clang 20.1.4 ]
executable: /home/arthur/probabl/skore/skore/.mlflow-venv/bin/python
   machine: Linux-6.17.0-14-generic-x86_64-with-glibc2.39

Python dependencies:
        skore: 0.0.0+unknown
          pip: 26.0.1
       jinja2: 3.1.6
       joblib: 1.5.3
   matplotlib: 3.10.8
        numpy: 2.4.2
       pandas: 2.3.3
       plotly: 6.5.1
         rich: 14.3.3
 scikit-learn: 1.8.0
      seaborn: 0.13.2
 skore[local]: None
        skrub: 0.7.1
skore-hub-project: None
```

### PR DESCRIPTION
**Pull request by auguste-probabl**
#### Change description

Closes https://github.com/probabl-ai/skore/issues/2546

#### Contribution checklist

- [x] Unit tests were added or updated
- [x] The code passes our style conventions (you can check this by running pre-commit on your code
with `pre-commit run --all-files`)
- [ ] All the tests pass (please test locally before pushing)
- [ ] The documentation builds and renders properly (if it does, our bot will add a comment linking
to a preview of the documentation to review it visually)
- [x] All the commits in the PR are signed
- [x] The pull request title respects the Conventional Commits convention
#### AI usage disclosure

AI tools were involved for:
- [ ] Code generation (e.g., when writing an implementation or fixing a bug)
- [ ] Test/benchmark generation
- [ ] Documentation (including examples)
- [ ] Research and understanding

### CODE REVIEW

**Comment by GaetandeCast** [comment]
I think we should we resolve merge conflicts and merge this since it probably going to take time to implement `data_source="both"` for cross val across the board.

**Review by GaetandeCast — APPROVED** [review]
